### PR TITLE
EOS-26584: [K8S monitor] Filter alerts based on multiple labels

### PR DESCRIPTION
**Pull request by akash2144**
Signed-off-by: Akash Dudhane <akash.a.dudhane@seagate.com>

# Problem Statement
- [K8S monitor] Filter alerts based on multiple labels
- Design/Implement label based filter which allow filtering based on multiple labels instead of a single label.

# Design
-  With the help of **in** operator we can query events for multiple pods_labels
   kwargs['label_selector'] = f'app in (dummy-node, test-node, hello-node)'

# Coding
-  [x] Coding conventions are followed and code is consistent

# Testing 
- [x] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM

# Review Checklist 
- [x] PR is self reviewed
- [x] JIRA number/GitHub Issue added to PR
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained
- [ ] Is there a change in filename/package/module or signature? [Y/N]: 
- [ ] If yes for above point, is a notification sent to all other cortx components? [Y/N]
- [ ] Side effects on other features (deployment/upgrade)? [Y/N]
- [ ] Dependencies on other component(s)? [Y/N]
-     If yes for above point, post link to the corresponding PR.

# Review Checklist 
- [ ] Is perfline test run and the report with and without the changes updated in the PR? [Y/N]: 

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

## Unit Test Details
<pre>
  <code>
    <h3> 1. Run ha config command inside k8s container & confirmed that pod_labels are populated in ha.conf file
    <h4>    [root@cortx-control-headless-svc /]# /opt/seagate/cortx/ha/bin/ha_setup config --config yaml:///etc/cortx/cluster.conf --services fault_tolerance
     config command is successful.
     [root@cortx-control-headless-svc /]# cat /etc/cortx/ha/ha.conf
     EVENT_MANAGER:
            consumer_group: health_monitor
            consumer_id: '1'
            message_type: health_events
            producer_id: system_health
     LOG:
            level: INFO
            path: /var/log/seagate/cortx/ha
     MONITOR:
            message_type: k8s_event
            producer_id: k8s_monitor
     consul_config:
            endpoint: tcp://consul-server.default.svc.cluster.local:8500
     data_pod_label:
     - cortx-data
     - cortx-server
     event_topic: hare
     <h3> 2. Run monitor inside k8s container & confirmed that it shows events for those pod_labels which are specified in ha.conf file.
     <h4>    Labels associated with pods are:
               Name: cortx-data-pod-ssc-vm-g3-rhev4-2663-7b9bbb896b-8zv42    Labels: app=cortx-data-pod-ssc-vm-g3-rhev4-2663  name=cortx-data
               Name: cortx-data-pod-ssc-vm-g3-rhev4-2664-5b7677897-4999v    Labels: app=cortx-data-pod-ssc-vm-g3-rhev4-2664  name=cortx-data
               Name: cortx-data-pod-ssc-vm-g3-rhev4-2664-5b7677897-4999v    Labels: app=cortx-data-pod-ssc-vm-g3-rhev4-2665  name=cortx-data
    Events seen in logs:
   Sending alert on message bus {'_resource_type': 'k8s:node', '_resource_name': 'ssc-vm-g3-rhev4-2663.colo.seagate.com', '_event_type': 'online', '_k8s_container': None, '_pod': None, '_node': None, '_is_status': True, '_timestamp': '1638181654'}
    Sending alert on message bus {'_resource_type': 'k8s:node', '_resource_name': 'ssc-vm-g3-rhev4-2664.colo.seagate.com', '_event_type': 'online', '_k8s_container': None, '_pod': None, '_node': None, '_is_status': True, '_timestamp': '1638181654'}
    Sending alert on message bus {'_resource_type': 'k8s:node', '_resource_name': 'ssc-vm-g3-rhev4-2665.colo.seagate.com', '_event_type': 'online', '_k8s_container': None, '_pod': None, '_node': None, '_is_status': True, '_timestamp': '1638181654'}
    Sending alert on message bus {'_resource_type': 'k8s:pod', '_resource_name': 'cortx-data-pod-ssc-vm-g3-rhev4-2665-97cb5488-nxslk', '_event_type': 'online', '_k8s_container': None, '_pod': None, '_node': 'ssc-vm-g3-rhev4-2665.colo.seagate.com', '_is_status': True, '_timestamp': '1638181654'}
    Sending alert on message bus {'_resource_type': 'k8s:pod', '_resource_name': 'cortx-data-pod-ssc-vm-g3-rhev4-2664-5b7677897-4999v', '_event_type': 'online', '_k8s_container': None, '_pod': None, '_node': 'ssc-vm-g3-rhev4-2664.colo.seagate.com', '_is_status': True, '_timestamp': '1638181654'}
    Sending alert on message bus {'_resource_type': 'k8s:pod', '_resource_name': 'cortx-data-pod-ssc-vm-g3-rhev4-2663-7b9bbb896b-8zv42', '_event_type': 'online', '_k8s_container': None, '_pod': None, '_node': 'ssc-vm-g3-rhev4-2663.colo.seagate.com', '_is_status': True, '_timestamp': '1638181655'}
  </code>
</pre>

### CODE REVIEW

**Comment by ArchanaLimaye** [comment]
Please explain the actual problem we are addressing here in "Problem Statement"

**Comment by ArchanaLimaye** [comment]
In unit test details, can you also share the labels associated with each pod for which alerts are seen

**Comment by akash2144** [comment]
> In unit test details, can you also share the labels associated with each pod for which alerts are seen

Added Labels with pod name in test details.

**Review by ArchanaLimaye — APPROVED** [review]

**Comment by ArchanaLimaye** [comment]
can you rebase to latest branch

**Comment by ArchanaLimaye** [comment]
Also verify that if single label is configured, that also works fine (if not already tested)

**Comment by akash2144** [comment]
> Also verify that if single label is configured, that also works fine (if not already tested)

Testing is done for this scenario.

File: ha/k8s_setup/ha_setup.py [hunk 1]
```diff
@@ -178,7 +178,7 @@ def process(self):
             # Ref ticket EOS-25694
             data_pod_label = Conf.get(self._index, f'cortx{_DELIM}common{_DELIM}product_release')
             # TBD delete once data_pod_label is avilable from confstore
-            data_pod_label = 'cortx-data'
+            data_pod_label = ['cortx-data', 'cortx-server']
```
Comment (ajaysrivas) [inline ha/k8s_setup/ha_setup.py:181]: Currently these are hard coded but I assume that this information will come from Conf Store eventually?

**Review by ajaysrivas — APPROVED** [review]

**Pull request merged**